### ISSUE TITLE
LSTM recurrent dropout changes variance only slightly

### ISSUE THREAD

**Issue by saraheberhard** [keras-team/keras#10838]
I am running a two-input-model with Dense layers and a LSTM layer. I am trying to run @yaringal s MC Dropout. For the LSTM layer I am using Keras' recurrent dropout. But it seems like only the Dropout of the Dense layers is working as I am getting only very small variance.

My model Looks like that:

```
input1 = Input(shape = ((train_X1.shape[1], train_X1.shape[2])))
input2 = Input(shape = (train_X2.shape[1],))

y = Dense(10, activation = 'relu')(input2)
y = Dropout(0.5)(y)
NN_out = Dense(1, name = 'NN_out')(y)
NN_out = Dropout(0.5)(NN_out)
NN_out = Reshape((1,1))(NN_out)
con = concatenate([input1, NN_out], axis = -1)
lstm_out = LSTM(20, recurrent_dropout = 0.5)(con)
out = Dense(11)(lstm_out)

model = Model(inputs = [input1, input2], outputs = out)
```

To get MC Dropout I am doing the following:

```
import keras.backend as K

class KerasDropoutPrediction(object):
    def __init__(self,model):
        self.f = K.function([model.inputs[0], model.inputs[1], K.learning_phase()], 
                            [model.layers[-1].output])

    def predict(self, x1, x2, n_iter = 100):
        result = []
        for _ in range(n_iter):
            result.append(self.f([x1,x2,1]))
        result = np.array(result)
        return result

kdp = KerasDropoutPrediction(model)
result = kdp.predict(test_X1, test_X2, n_iter = 100)
```

I am using Tensorflow backend.
Does anybody know why this is not working?

**Comment by yitelee** [comment]
I had same problem here, and found it was related to the input shape of X_train you prepared. For example:
`model.add(LSTM(50, input_shape=(input_timesteps, input_dim), recurrent_dropout=0.1))`
If input_timesteps = 1, recurrent_dropout has no effect, you feel no difference of the result even you set it to 0.99
If input_timesteps = 10, varying recurrent_dropout will show its effect.
